site/src/components/address-input/address-input.tsx

```diff
@@ -31,252 +23,97 @@ export class AddressInput {
-         // see: https://smartystreets.com/docs/cloud/us-autocomplete-api#pro-secondary-expansion
-         "selected": selected == null || selected.entries < 2 ? ""
-            : `${selected.street_line} ${selected.secondary} (${selected.entries}) ${selected.city} ${selected.state} ${selected.zipcode}`,
-         "include_only_states": this.stateValue || "",
-      } ), {
-         method: "GET",
-      } )
-         .then( response => response.json().then( ( data: USAutocomplete.QueryResult ) => {
-            let suggestions = ( data.suggestions || [] ).map( ( suggestion, i: number ) => {
-               return {
-                  suggestion: suggestion,
-                  value: i,
-                  text: suggestion.street_line
-                     + ( suggestion.secondary ? " " + suggestion.secondary : "" )
-                     + ( suggestion.entries > 1 ? " (" + suggestion.entries + " more entries)" : "" )
-                     + " " + suggestion.city + ", " + suggestion.state + " " + suggestion.zipcode,
-               };
-            } );
-            this.addressOptions = suggestions;
-         } ) )
-         .catch( err => console.log( "smartystreets error", err ) );
+         { AddressInput.possiblyHiddenSelect("State", "state", this.stateValue, this.stateOptions)}
+
+      </Host> );
    }
 
-   /**
-    * Try to ge tmore detailed street data given the user's selection
-    */
-   private getSingleAddressData( address: USAutocomplete.Suggestion ) {
-      fetch( "https://us-street.api.smartystreets.com/street-address?" + toQueryString( {
-         "auth-id": this.smartyStreetsApiKey,
-         "street": address.street_line,
-         "city": address.city,
-         "state": address.state,
-      } ), {
-         method: "GET",
-      } )
-         .then( response => response.json().then( ( data: USStreetAddress.QueryResult ) => {
-            const result = data[0];
-            if( result != null ) {
-               this.cityValue = result.components?.city_name || "";
-               this.stateValue = result.components?.state_abbreviation || "";
-               this.zipValue = result.components?.zipcode || "";
-               this.address2Value = result.delivery_line_2 || "";
-               this.countyValue = result.metadata?.county_name || "";
-            }
-         } ) )
-         .catch( err => console.log( "smartystreets error", err ) );
+   private static possiblyHiddenSelect(
+      fieldLabel: string,
+      name: string,
+      selected: string,
+      options: Map<string, string> | Set<string>
+   ) {
+      const labelClass = options.size <= 1 ? { class: "hide" } : {};
+      let optionTags = new Array<HTMLElement>();
+
+      for(let [value, label] of options.entries()) {
+         optionTags.push(<option value={value} selected={selected === value}>{ label }</option>);
+      }
+
+      return (
+         <label {...labelClass}>
+            { fieldLabel }<span class="ak-required-flag">*</span>
```

Comment:
You can remove this `class="ak-required-flag"` it's a holdover from the previous design